cases_new/arithmetic/add.yaml

```diff
@@ -0,0 +1,47 @@
+base_uri: https://github.com/substrait-io/substrait/blob/main/extensions/substrait/extensions/functions_arithmetic.yaml
+function: add
+cases:
+- group: basic
+  description: Basic examples without any special cases
+  tests:
+  - (120|i8, 5|i8): 125|i8
+  - (100|i16, 100|i16): 200|i16
+  - (30000|i32, 30000|i32): 60000|i32
+  - (2000000000|i64, 2000000000|i64): 4000000000|i64
```

Comment:
The more time I spend with this, I'm inclined to forgo using yaml. If we have trivial parsing, we might as well make the format as clear as possible. How about something like this (rough spec):

```
doc         := <version>
               (<include>)*
               ((<test_group>)?(<test_case>)+\n)+            
version     := ### SUBSTRAIT_SCALAR_TEST: <test_library_version>
include     := ### SUBSTRAIT_INCLUDE: <uri>
test_group  := # <description>
test_case   := <function>(<arguments>) ([<options>])? = <result> (#<description>)?
description := string
function    := string
arguments   := <argument>, <argument>, ... <argument>
argument    := <literal>
literal     := <name>::<datatype>
result      := SUBSTRAIT_ERROR | <literal>
options     := <optLiteral>, <optLiteral>, ... <optLiteral>
optLiteral  := <option_name>:<option_value>

```

So an example might look like:

```
### SUBSTRAIT_SCALAR_TEST:V1
### SUBSTRAIT_INCLUDE: /extensions/functions_aggregate_approx.yaml

# Common Maths
add(126::i8, 1::i8) = 127::i8

# Arithmetic Overflow Tests
add(127::i8, 1::i8) [overflow:ERROR] = <!ERROR>  #check overflow
```

We're describing two test groups "Common Maths" and "Arithmetic Overflow Tests". Each has one test case. The test case in the second group has a name whereas case in the first one does not.